### PR TITLE
fix

### PR DESCRIPTION
**Pull request by zhuoshen617**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,93 +0,0 @@
...
-function getRankings(prevStockList, currStockList) {
-  let rankingChanges = [];
-  let prevPositions = [];
-  let currInPrevPositions = [];
-
-  for (prevStock of prevStockList) {
-    prevPositions.push(prevStock["ticker"]);
-  }
-
-  for (let i=0; i<10; i++) {
```
Comment (zhuoshen617) [inline app.js:40]: for (let i = 0; i < 10; i++) {
}

File: app.js [hunk 2]
```diff
@@ -1,93 +0,0 @@
...
-const Count = require("./models/visitorCount");
-
-const dbUrl = process.env.DB_URL;
-
-mongoose.connect(dbUrl, { useNewUrlParser: true }, { useUnifiedTopology: true })
-    .then(() => {
-        console.log("MONGO CONNECTION OPEN")
-    })
-    .catch(err => {
-        console.log("MONGO CONNECTION ERROR")
```
Comment (zhuoshen617) [inline app.js:20]: console.error("MONGO CONNECTION ERROR:", err)

File: app.js [hunk 3]
```diff
@@ -1,93 +0,0 @@
...
-    
-  }
-  return rankingChanges;
-}
-
-app.get("/", async (req, res) => {
-  const latestCount = await Count.find().sort({_id: -1}).limit(1);
-  const visitorCount = latestCount[0].count + 1; 
-  const filter = {count: visitorCount - 1};
-  const update = {count : visitorCount};
```
Comment (zhuoshen617) [inline app.js:60]: remove space after count

File: app.js [hunk 4]
```diff
@@ -1,93 +0,0 @@
...
-    else {
-      rankingChanges.push({"ticker": currStockList[i]["ticker"], "change": currStockList[i]["rank"]-currInPrevPositions[i]});  
-    }
-    
-  }
-  return rankingChanges;
-}
-
-app.get("/", async (req, res) => {
-  const latestCount = await Count.find().sort({_id: -1}).limit(1);
```
Comment (zhuoshen617) [inline app.js:57]: Wrap this in another function called updateViewCount and do this after the loading ticker logic. 

Also if this failed, you should still display the main page. So this is a non-blocking error.

Do some research to make this code fail, like accessing a table that doesn't exist. Make sure the page still loads.

File: app.js [hunk 5]
```diff
@@ -1,93 +0,0 @@
...
-  const visitorCount = latestCount[0].count + 1; 
-  const filter = {count: visitorCount - 1};
-  const update = {count : visitorCount};
-  let doc = await Count.findOneAndUpdate(filter, update, {upsert: true, sort: {created: -1}});
-
-  const latest = await Stock.find().sort({timeStamp:-1}).limit(1);
```
Comment (zhuoshen617) [inline app.js:63]: Wrap this into another function called loadTicker or something to make this function really simple and easier to read.

File: app.js [hunk 6]
```diff
@@ -1,93 +0,0 @@
...
-  const latestTimeStamp = latest[0].timeStamp;
-  const stockList  = await Stock.find({timeStamp: latestTimeStamp}).exec();
-  const prev = await Stock.find({timeStamp: {$lte: latestTimeStamp - 2*60*60*1000}}).sort({timeStamp:-1}).limit(1);
```
Comment (zhuoshen617) [inline app.js:66]: define 2*60*60*1000 in a constant.

File: app.js [hunk 7]
```diff
@@ -1,93 +0,0 @@
...
-  const prevTimeStamp = prev[0].timeStamp;
-  const prevStockList = await Stock.find({timeStamp: prevTimeStamp}).exec();
-
-  let obj = {};
-
-  for (let i=0; i<stockList.length; i++) {
```
Comment (zhuoshen617) [inline app.js:72]: style.

File: app.js [hunk 8]
```diff
@@ -1,93 +0,0 @@
...
-    })
-
-app.set("views", path.join(__dirname,"views"));
-app.set("view engine", "ejs");
-
-app.use(express.urlencoded ({extended:true}));
-app.use(express.static(__dirname));
-
-
-function getRankings(prevStockList, currStockList) {
```
Comment (zhuoshen617) [inline app.js:31]: getRankingChanges

File: app.js (hunk 1 above)
Comment (LilyXueLi) [reply app.js:40]: ?? to be discussed

File: app.js (hunk 2 above)
Comment (LilyXueLi) [reply app.js:20]: done

File: app.js (hunk 8 above)
Comment (LilyXueLi) [reply app.js:31]: done

File: app.js (hunk 1 above)
Comment (zhuoshen617) [reply app.js:40]: space around i=0

File: app.js (hunk 3 above)
Comment (LilyXueLi) [reply app.js:60]: done

**Pull request closed without merging**